### DISCUSSION TITLE
Option to disable regex in `replace` filter

### DISCUSSION THREAD

**Discussion by ghost** [craftcms/cms#13642] [Ideas] [answered]
Hi Folks,

Following on from [an issue](https://github.com/craftcms/cms/issues/13638) I posted yesterday, I have been thinking about whether it is a good idea to have the `replace` filter acting as both `str_replace` and `preg_replace` (which is how it currently works).

The impetus for the above issue was actually due to some existing code breaking after updating to Craft 4.5.x

The goal of the existing code was to remove a path segment such as `/a/` from a URL, *including* the surrounding slashes. The code looked like something like this:

```
{% set url = 'https://example.com/a/' %}

{% set urlWithoutPath = url|replace({'/a/': ''}) %}
```

Previously that would set `urlWithoutPath` to `'https://example.com'`.

Since updating to Craft 4.5, this instead removes all instances of the `'a'` character from the string, thus resulting in `urlWithoutPath` being set to `https://exmple.com//`.

While this is technically the expected behaviour since the new version supports a regex, it is a bit of a breaking change in a sense, since the `replace` interface is no longer backwards compatible with our (valid) use-case.

The pattern kindly provided by @wsydney76 in that issue, `/\\/a\\//`, does work, but seems quite an inelegant/ugly solution when a `str_replace` would simply need `/a/`. 

I've been trying to think about ways of preventing similar unexpected behaviour in future. Perhaps an option on the `replace` filter to disable the regex matching would help prevent confusion, and also avoid us having to use hard-to-read regex patterns.

Thoughts?

**Comment by brandonkelly** [comment, marked as answer]
Sorry, that was an unintended change in behavior.

I’ve [just added](https://github.com/craftcms/cms/commit/361c5fc085bde0bc55b64ab95e9ed5fc1ca72287) a new `regex` argument to the `|replace` filter for the next release, which you can set to `false` to explicitly disable regex parsing:

```twig
{% set url = 'https://example.com/a/' %}

{% set urlWithoutPath = url|replace({'/a/': ''}, regex=false) %}
```

**Reply by ghost** [reply]
Perfect, thanks very much Brandon!

**Reply by brandonkelly** [reply]
Craft 4.5.4 is out with that change.
